lib/tasks/redis.rake

```diff
@@ -0,0 +1,22 @@
+require "redis"
+
+namespace :redis do
+  desc "Make redis a slave of production redis"
+  task :make_slave => :environment do
+    redis = Redis.new(url: ENV.fetch('REDIS_URL'), :driver => :hiredis)
+    redis.slaveof(ENV.fetch('PRODUCTION_REDIS_HOST'), ENV.fetch('PRODUCTION_REDIS_PORT'))
+    redis.config('set', 'masterauth', ENV.fetch('PRODUCTION_REDIS_PASSWORD'))
+    redis.config('set', 'slave-read-only', 'no')
```

Comment:
I'm not entirely sure if we want the slave to be writable because theoretically they should all just sync from the primary, since only the production instance will retrieve data from CMC.

Reply:
From the documentation, it allows you to write locally and then it'll just be overwritten upon update. This makes it so that rails doesn't throw an error when it wants to write to redis.

> You may wonder why it is possible to revert the read-only setting and have slave instances that can be targeted by write operations. While those writes will be discarded if the slave and the master resynchronize or if the slave is restarted, there are a few legitimate use case for storing ephemeral data in writable slaves.

> Also note that since Redis 4.0 slave writes are only local, and are not propagated to sub-slaves attached to the instance. Sub slaves instead will always receive the replication stream identical to the one sent by the top-level master to the intermediate slaves. So for example in the following setup: